numpyro/infer_util.py

```diff
@@ -56,3 +53,10 @@ def transform_fn(transforms, params, invert=False):
         transforms = {k: v.inv for k, v in transforms.items()}
     return {k: transforms[k](v) if k in transforms else v
             for k, v in params.items()}
+
+
+def constrain_fn(model, model_args, model_kwargs, transforms, params):
```

Comment:
Same as above - we should document that this constrains to base values, and perhaps change the name or add an optional kwarg like `use_base_dist_params=True`. We can punt this to later PRs after the refactoring is done.